service/src/test/java/uk/gov/hmcts/reform/fpl/handlers/PopulateStandardDirectionsHandlerTest.java

```diff
@@ -20,295 +17,174 @@
-
-        handler.populateStandardDirections(new PopulateStandardDirectionsEvent(callbackRequest, requestData));
-
-        verifyCoreCaseDataApiIsCalledWithCorrectParameters();
-
-        List<Direction> directions = localAuthorityDirections(getCaseData());
-
-        assertThat(directions).containsOnly(
-            expectedDirection(DIRECTION_TEXT, MONDAY_DATE_AT_NOON.minusDays(4)),
-            expectedDirection(DIRECTION_TEXT, MONDAY_DATE_AT_NOON.minusDays(5)));
-    }
-
-    @Test
-    void shouldPopulateStandardDirectionsWhenPopulatedDisplayInConfiguration() throws IOException {
-        callbackRequest = getCallbackRequestWithCustomHearingOnMonday();
-
-        given(startPopulateStandardDirectionsEvent()).willReturn(getStartEventResponse(callbackRequest));
-
-        given(ordersLookupService.getStandardDirectionOrder())
-            .willReturn(getOrderDefinition(DIRECTION_TEXT, SAME_DAY_AS_HEARING));
-
-        handler.populateStandardDirections(new PopulateStandardDirectionsEvent(callbackRequest, requestData));
-
-        verifyCoreCaseDataApiIsCalledWithCorrectParameters();
-
-        assertThat(caseDataContent.getValue()).isEqualTo(expectedCaseDataContent(callbackRequest));
-        assertThat(localAuthorityDirections(getCaseData()))
-            .containsOnly(expectedDirection(DIRECTION_TEXT, MONDAY_DATE_AT_NOON));
-    }
-
-    @Test
-    void shouldPopulateStandardDirectionsWhenNullDeltaValueInConfiguration() throws IOException {
-        given(startPopulateStandardDirectionsEvent()).willReturn(getStartEventResponse(callbackRequest));
-
-        given(ordersLookupService.getStandardDirectionOrder())
-            .willReturn(getOrderDefinition(DIRECTION_TEXT, null));
-
-        handler.populateStandardDirections(new PopulateStandardDirectionsEvent(callbackRequest, requestData));
 
-        verifyCoreCaseDataApiIsCalledWithCorrectParameters();
+        when(calendarService.getWorkingDayFrom(eq(DATE), eq(parseInt(TWO_DAYS_BEFORE_HEARING))))
+            .thenReturn(DATE.minusDays(2));
+        when(calendarService.getWorkingDayFrom(eq(DATE), eq(parseInt(THREE_DAYS_BEFORE_HEARING))))
+            .thenReturn(DATE.minusDays(3));
 
-        assertThat(caseDataContent.getValue()).isEqualTo(expectedCaseDataContent(callbackRequest));
-        assertThat(localAuthorityDirections(getCaseData()))
-            .containsOnly(expectedDirection(DIRECTION_TEXT, null));
     }
 
     @Test
-    void shouldPopulateStandardDirectionsWhenTextContainsSpecialCharacters() throws IOException {
-        callbackRequest = getCallbackRequestWithCustomHearingOnMonday();
-
-        given(startPopulateStandardDirectionsEvent()).willReturn(getStartEventResponse(callbackRequest));
-
-        given(ordersLookupService.getStandardDirectionOrder())
-            .willReturn(getOrderDefinition(DIRECTION_TEXT_SPECIAL_CHARACTERS, SAME_DAY_AS_HEARING));
+    void shouldAddDirectionsToCaseData() throws IOException {
```

Comment:
I think it's too detailed and it's not even unit test anymore, because we're not isolating it. We should mock dependencies instead of autowiring them. The only responsibilities of PopulateStandardDirectionsHandler that should be tested are:
- hearingBookingService call
- standardDirectionsService call with data returned from the above (we don't really care if the data returned here is valid, this should be verified in standardDirectionsService unit test)
- commonDirectionService call with data returned from standardDirectionsService
- case data is populated with data returned from commonDirectionService (again, this doesn't need to be valid data from end-to-end perspective)
- ccd event is triggered

Thoughts? I'm happy to merge it like this, because it was already too complex before. I was doing these changes on my branch anyway, so I'm fine with leaving it as it is in your PR.

Reply:
sure, so I guess mocking out hearing service and standard directions service call will solve your comment?

Reply:
I'd mock commonDirectionService as well, just to have it as isolated as much as possible.